### DISCUSSION TITLE
Understanding the BlueRetro programming language.

### DISCUSSION THREAD

**Discussion by SinisterSticks** [darthcloud/BlueRetro#866] [General]
On video today, I thought I programmed my RetroOnyx Virtual Boy Bluetooth adapter correctly.  I talked my way through it long form.

The programm ing is for Hyper Fighting on VB (BTW, if you want to see a real VB in 3d without a kink in the neck, get out some cheap paper red and cyan 3d glasses and watch, and be impressed.)

The video is on twitch.tv/tripletopper from 8:35 PM  New York Time today (Dec 21, 2023) .  You coukd see "talk out the logic". Apparently, only the Heavy Kick works.

Can you watch the video and see whether it's human error or machine error?  And if machine error, you can probably better describe it than an outsider like me.

BTW if my joystick looks weird, it's a SinisterSticks.com stick (full disclosure my own URL), is designed to be ambidextrous, and operates off an Xbox Adaptive Controller.

**Comment by SinisterSticks** [comment]
Hello, has anyone reviewed my video to see it my usage of the blueretro.io website is correct?

For some strange reason, my blueretro programming doesn't have any attack work except Heavy Kick with my Xbox Adaptive Controller hooked up to a fight stick.

Any suggestion of what I have done wrong in programming? Any user error?

**Comment by darthcloud** [comment]
The config look fine to me.

For troubleshooting:
* 1
You could try to reload the default config "Default gamepad/keyboard" from the preset page:
https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-BLE-Web-Config-User-Manual#32---mapping-config
Then can you confirm using a regular controller (like PS4, PS5 or Xbox regular) that all the buttons works using the default config.

* 2
Make your custom config and double check again with a regular controller.

If everything worked at step 1 and 2 then it's a problem either in your setup or with the Xbox adaptive.

**Reply by SinisterSticks** [reply]
I have an interesting theory.

I noticed that the default is to take the right analog stick and define that as the right stick on Virtual Boy.  The right stick starts as analog and gets mapped to the VB which is digital.  

But on my fight stick, the ABXY of the Xbox Adaptive start as digital and are supposed to be digital, but maybe it is expecting an analog input concerted to digital output.

The right trigger is thought of as analog on the Xbox Adaptive, but a digital button can actuate the analog control 100%.

The right bumper is digital, but is mapped to the R.

I can't tell if the native VB controls have digital or analog buttons.  But the only button that works right is the right trigger which maps heavy kick to heavy kick correctly.

Maybe it's an analog/digital issue.  I had a similar problem with the digital buttons mapped to analog buttons on a Game Cube when I played Total Carnage with Sega Master System Twin Sticks.

Just wondering if that would cause an issue.

**Reply by darthcloud** [reply]
Virtual Boy is all digital buttons.

BlueRetro can map anything to anything. digital to analog, digital to digital, analog to digital and analog to analog.

That's a core concept of BlueRetro, so that isn't the issue.

**Comment by SinisterSticks** [comment]
I think I have partially figured it out because I read something that didn't do the first time when I set it up on my computer, but noticed on my phone, was that you have to reset the Bluetooth adapter in order for the settings to gel.

Is it correct that this is the proper order for fresh start of hooking up an Xbox adaptive to a retronyx vbbt?

(By the way I posted it on a new thing in general discussions but then realized might as well bring it back up in the original discussion so we could refer to it easier)
1. Clear Xbox adaptive of all memory by holding down the OS button and the sync button together for 20 seconds.
2. Clear Virtual Boy Bluetooth by holding down the reset button( left button when inserted upside down like my consoleized VB is) for 10 seconds.
3. Now I should scan for my VBBT on my android phone.
4. Program
5. Reset VBBT
6. Sync
7.  Play

Am I doing something in the wrong order or am I doing in the right order and am I doing the scanning correctly?

**Comment by SinisterSticks** [comment]
I tried doing everything in that order, and the only things that work are the L trigger and R the triggers.

For some strange reason when ABXY as right stick north south east west it doesn't gel even after with a reset.   

You saw my video I changed the appropriate things.  What if the first column is supposed to be untouched and only the second column change then why is the first column changeable because I changed my buttons on the first column to correspond with the existing right control on the other side of the column.

Does it work better if you just keep the left the same and just adjust the right or vice versa?  It doesn't work mixing both..
